nlmod/util.py

```diff
@@ -1054,3 +1058,191 @@ def _get_value_from_ds_datavar(
             value = value.values
 
     return value
+
+
+def add_info_to_gdf(
```

Comment:
name could be more descriptive, maybe `gdf_intersection_join`. And a docstring woulde be nice :)